File: MEAN/EVA-Beheer/src/server/db/models/challenge.js
var mongoose = require('mongoose');
var Schema = mongoose.Schema;

var ChallengeSchema = new Schema({
  titel: { type: String, required : true},
  omschrijving: { type: String, required : true},
  dag: { type: Number, required : true},
	restaurant : {
    naam: String,
  	adres: {
  		straat: String,
      huisnummer: String,
      stad: String,
      postcode: String
  	},
    telefoon: String
  },
  gerecht : {
      naam: String,
      omschrijving: String,
      categorie: {
          naam: String
      },
      allergenen: [{
          naam: String,
          selected: Boolean
      }]
  },
  reward : String
});

module.exports = mongoose.model('Challenge', ChallengeSchema);

File: Android/EVA/app/src/main/java/com/projecten3/eva/Model/Challenge.java
package com.projecten3.eva.Model;

public class Challenge {
    public String id;
    public String titel;
    public String omschrijving;
    public int dag;
    public String reward;
    public Gerechten gerecht;
    public Restaurant restaurant;

    public Challenge(String id, String titel, String omschrijving, int dag, String reward, Gerechten gerecht, Restaurant restaurant) {
        this.id = id;
        this.titel = titel;
        this.omschrijving = omschrijving;
        this.dag = dag;
        this.reward = reward;
        this.gerecht = gerecht;
        this.restaurant = restaurant;
    }

    public String getId() {
        return id;
    }

    public String getTitel() {
        return titel;
    }

    public String getOmschrijving() {
        return omschrijving;
    }

    public int getDag() {
        return dag;
    }

    public String getReward() {
        return reward;
    }

    public Gerechten getGerecht() {
        return gerecht;
    }

    public Restaurant getRestaurant() {
        return restaurant;
    }
}
